lower.ts

```diff
@@ -21,7 +21,39 @@ function generateName(base : string) : string {
 //   return [name, {tag: "label", a: a, name: name}];
 // }
 
+export function generateVtable(p : AST.Program<Type>, env : GlobalEnv) {
+  var vtable : Array<string> = [];
+  var classIndexes = new Map(); // stores the start and end index of the class in vtable
+  var methodIndex = 0;
+  p.classes.forEach(cls => {
+    if (cls.super[0]!=="object") {
+      const superClassIndexes = classIndexes.get(cls.super[0])
+      var superClassVtable = vtable.slice(superClassIndexes[0], superClassIndexes[1])
+      cls.methods.filter(m => m.name !== "__init__").forEach(m => {
+        const methodOffset = env.classes.get(cls.name)[1].get(m.name)
+        if (methodOffset > superClassVtable.length) {
+          superClassVtable.push(`$${cls.name}$${m.name}`)
+        } else {
+          superClassVtable[methodOffset] = `$${cls.name}$${m.name}`;
+        }
+      })
+      classIndexes.set(cls.name, [vtable.length, vtable.length +superClassVtable.length])
+      vtable = [...vtable , ...superClassVtable]
+    } else {
+      // add methods directly and increment methodIndex
+      classIndexes.set(cls.name, [vtable.length, vtable.length + cls.methods.length - 1])
+      cls.methods.filter(m => m.name !== "__init__").forEach(m => {
```

Comment:
Why leave out `__init__`? Can't it still be called like other methods?

Reply:
Yes, we can do that. 

We initially thought it might be simpler to keep the constructors separate as simple function calls. But after reviewing it again now, we agree that it would be easier to handle it like methods since we won't have to filter and handle `__init__` separately every time we need to do something with class methods. 

The code is updated now.